solr/solrj/src/java/org/apache/solr/client/solrj/cloud/autoscaling/AutoScalingConfig.java

```diff
@@ -587,11 +585,11 @@ public boolean equals(Object o) {
     if (!getTriggerListenerConfigs().equals(that.getTriggerListenerConfigs())) return false;
     return getProperties().equals(that.getProperties());
   }
-//  @Override
-//  public int hashCode() {
-//    throw new UnsupportedOperationException("TODO unimplemented");
-//  }
 
+  @Override
+  public int hashCode() {
+    return Objects.hash(policy, zkVersion);
```

Comment:
`equals()` does not compare `zkVersion`, so `hashCode()` should not be based on it either.

Reply:
Looks like this just needs to be a hash on getPolicy(). I don't think it can even be the member var policy since getPolicy can change it.

Reply:
Yeah, thought about including the other props, but I think we're OK with just getPolicy(). 

Like I mentioned before, though, I'm trying to limit scope here so I'm going to leave the getPolicy stuff as it is.